docs/reinterpreting_effects.md

```diff
@@ -0,0 +1,359 @@
+# Reinterpreting effects
+
+This article assumes you are already familiar with [defining effects and their handlers](defining_effects.md).
+
+One of the nice aspects of effects is that they can support multiple effect handlers. Effects only specify actions, they don't actually perform them. Therefore, it's possible to "reinterpret" effects. There are multiple senses in which an effect can be reinterpreted:
+
+- Implementing an effect in terms of other effects. "Stacking" effects is a powerful tool for cleanly dividing implementations into the relevant abstraction layers with minimal leakage of implementation details.
```

Comment:
This is what I typically think of as “reinterpreting;” I think I picked up that specific nomenclature from @lexi-lambda’s `freer-simple`.